static/assets.js

```diff
@@ -36,4 +36,47 @@ function hideEmptySections() {
         }
         section.style.display = 'none'
     })
-}
\ No newline at end of file
+}
+
+//  ------------    Version Filtering
+function normalize_version(raw_version) {
+    let version = raw_version
+        .replace(/^[^\d]+/, '')
+        .replace(/[^\d]+$/, '');
+    return version ? Array.from({ ...version.split('.'), length: 3 }, (v, i) => v ?? 0).join('.') : null;
+}
+
+let versionsQuery = document.querySelectorAll('.asset-card .asset-card__tags .asset-card__bevy-versions .asset-card__tag');
+let versions = [...new Set([...versionsQuery]
+    .map(item => normalize_version(item.innerText))
+    .filter(i => i)
+)];
+
+
+let versionsSelect = document.querySelector('#assets-filter');
+if (versionsSelect) {
+    versions.map(i => {
+        var opt = document.createElement('option');
+        opt.value = i;
+        opt.innerHTML = i;
+        versionsSelect.appendChild(opt);
+    })
+}
+
+document
+    .querySelector('#assets-filter')
+    .addEventListener("change", (item) => {
+
+        for (const asset of document.querySelectorAll('.asset-card')) {
+            let tag = asset.querySelector('.asset-card__tags .asset-card__bevy-versions .asset-card__tag');
+            if (item.target.value === 'all_versions') {
+                asset.parentElement.style.display = 'block'
+            }
+            else if (tag) {
+                const searchMatch = item.target.value === normalize_version(tag.innerText);
```

Comment:
When anything but `All` is selected, this currently filters out things like "Official Bevy Examples" and "Bevy Cheatbook", which leaves the impression that these are out of date / do not support the latest version. A lot of useful stuff is filtered out the second a filter is enabled. Imo the cost of filtering out useful things is _very_ high, so we should ensure things aren't falling through the cracks here. This is a problem we'll want to solve before shipping this feature.

Reply:
```
 Imo the cost of filtering out useful things is very high
```
Agreed! The problem here is how to treat versions like  'git', '*', 'main' and empty string.
I can make that `*` will go into all filters. I can replace `main` to `*` wich will do same. But what to do with `git` and empty string`?

Reply:
Empty string I think should map to *. Do the `git` fields link to a specific commit?

Reply:
from what I see it is never more that just `git`
![asdadcasda](https://github.com/bevyengine/bevy-website/assets/30506/91e60a24-edd9-4f5c-8df7-5e9064f29075)

Reply:
Empty strings are mostly coming from `learning` category. I'm not sure where version should appear there but tutorials and videos are definitely might be outdated and would be good to have ability to filter them out

Reply:
Hmm. I think for this PR we should probably just mark them as *, and then fix that up later.

Reply:
@cart `When anything but All is selected, this currently filters out things like "Official Bevy Examples"` The problem with `Official Bevy Examples` is that it is explicitly marked as 0.12.0 so it gets filtered out when other versions selected